declarative/launcher/applicationmanager.cpp

```diff
@@ -59,20 +59,24 @@ QList<Application *> ApplicationManager::applications() const
 {
     QList<Application *> apps;
 
-    Q_FOREACH (Application *app, m_apps) {
+    Q_FOREACH (Application *app, m_apps.values()) {
```

Comment:
no need to call `.values()` here :)
> With QMap and QHash, foreach accesses the value component of the (key, value) pairs automatically, so you should not call values() on the container (it would generate an unnecessary copy, see below).

[(source)](http://doc.qt.io/qt-5/containers.html#the-foreach-keyword)

the use of `Q_FOREACH` is also discouraged
> Note: Since Qt 5.7, the use of this macro is discouraged. It will be removed in a future version of Qt. Please use C++11 range-for, possibly with qAsConst(), as needed.

[(source)](http://doc.qt.io/qt-5/qtglobal.html#Q_FOREACH)